### DISCUSSION TITLE
Add a "copy as import" code action

### DISCUSSION THREAD

**Discussion by brettcannon** [microsoft/pylance-release#2669] [Enhancement]
PyCharm has something like this, and it's incredibly useful when you want to import a function from your codebase e.g. in a Python REPL.

Let's say you have a file `mypackage/foo/bar.py` containing a `test` function. Then this action would copy `from mypackage.foo.bar import test`.

PyCharm copies `mypackage.foo.bar.test` but I think an actual import statement is more useful.

_Originally posted by @ThiefMaster in https://github.com/microsoft/vscode-python/discussions/16412_

**Comment by judej** [comment]
@thiefmaster, is there a description of the behavior or a gif that shows the functionality? thanks!

**Comment by ThiefMaster** [comment]
![pycharm-copy-reference](https://user-images.githubusercontent.com/179599/122809568-90d66e00-d2ce-11eb-94a2-dd1634fb2226.gif)

Note that the way PyCharm copies it is not perfect since it cannot be used as-is in Python code or a REPL (I wrote my own ipython extension for this when I still used PyCharm). But the gif shows pretty well how it works. The format how it's copied could be made configurable (placeholders for the symbol itself and the module path to it), or it could be always a "Copy as import" action resulting in a `from ... import ...` string as that's probably the most useful.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
